src/DocBlock/TypeExpression.php

```diff
@@ -78,24 +83,30 @@ final class TypeExpression
                 )
                 |
                 (?<class_constant> # class constants with optional wildcard, e.g.: `Foo::*`, `Foo::CONST_A`, `FOO::CONST_*`
-                    (?&name)::(\*|\w+\*?)
+                    (?&name)::\*?(?:(?&identifier)\*?)*
                 )
                 |
-                (?<array> # array expression, e.g.: `string[]`, `string[][]`
-                    (?&name)(\[\])+
-                )
-                |
-                (?<constant> # single constant value (case insensitive), e.g.: 1, `\'a\'`
+                (?<constant> # single constant value (case insensitive), e.g.: 1, -1.8E+6, `\'a\'`
                     (?i)
                     null | true | false
-                    | -?(?:\d+(?:\.\d*)?|\.\d+) # all sorts of numbers with or without minus, e.g.: 1, 1.1, 1., .1, -1
-                    | \'(?:[^\'\\\\]|\\\\.)*?\' | "(?:[^"\\\\]|\\\\.)*?"
+                    | [+-]?(?:
```

Comment:
Not tested for syntax correctness, but I would like to keep comment which was removed.

```suggestion
                    # all sorts of numbers: with or without sign / literal separator, in several numeric system,
                    # e.g.: 1, +1.1, 1., .1, -1, 123E+8, 123_456_789, 0x7Fb4, 0b0110, 0o777
                    | [+-]?(?:
```

Reply:
I had to revert, your regex comment has broken the regex, feel free to correct it and push to my PR branch

Reply:
That's why I explicitly stated that regex was not tested 😉 OK, let me checkout your branch.

Reply:
Funny, `sign / literal separator` was causing the problem because `/l` was treated as modifier 😅. I've improved the comment and tested `PhpCsFixer\Tests\AutoReview\DocumentationTest` locally, it worked. I hope other tests too 😉.

Reply:
👍modifiers are parsed before comments are stripped